dragonfly/entity/flammable.go

```diff
@@ -0,0 +1,15 @@
+package entity
+
+// Flammable is an interface for entities that can be set on fire.
+type Flammable interface {
+	// FireProof is whether the entity is currently fire proof.
+	FireProof() bool
+	// FireDamage deals fire damage to the entity.
+	FireDamage(amount float64)
+	// LavaDamage deals lava damage to the entity.
+	LavaDamage(amount float64)
+	// FireTicks returns duration of fire in ticks.
+	FireTicks() int
+	// SetFireTicks sets the player on fire for the specified duration in ticks.
+	SetFireTicks(ticks int)
```

Comment:
It's not very nice if we have to expose all these. A `SetOnFire(time.Duration)` method might be nicer, along with an `Extinguish()` method (and change `FireTicks()` to `OnFireDuration()`). As for the other methods, I'm guessing they're needed to get around circular import issues. Is there any way we can use an internal package to get around this?